### PR TITLE
✨ (signer-eth) [DSDK-601]: Add generic parser enums support

### PR DESCRIPTION
**Pull request by aussedatlo**
### 📝 Description

Add generic parser enums support.

- Context Module:
   - `HttpTransactionDataSource`: Return `ENUM` with `value`, `id` (used to map the enum used in the transaction) and correct payload (as each enum contain its own signature). transform `ENUM` field in `TRANSACTION_FIELD_DESCRIPTION` with an `ENUM`reference.
- Signer Eth:
   - `BuildTansactionGenericContextTask`: Filter `ENUM` context from all other `ClearSignContextSuccess` to use them differently in the next task
   - `ProvideTransactionFieldDescriptionTask`: Provide `ENUM` reference context if needed.

### ❓ Context

- **JIRA or GitHub link**: [DSDK-601]

- **Feature**:

### ✅ Checklist

Pull Requests must pass CI checks and undergo code review. Set the PR as Draft if it is not yet ready for review.

- [x] **Covered by automatic tests** 
- [x] **Changeset is provided** 
- [ ] **Impact of the changes:** 
  - _list of the changes_

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.

### CODE REVIEW

File: .changeset/curly-plants-doubt.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@ledgerhq/context-module": patch
+"@ledgerhq/device-signer-kit-ethereum": patch
```
Comment (jiyuzhuang) [inline .changeset/curly-plants-doubt.md:3]: [MAYBE] I think it's better to separate the packages due to the 4th point [here](https://ledger.slack.com/archives/C053YA9FHBR/p1733916669350689?thread_ts=1733916491.848089&cid=C053YA9FHBR)?

File: packages/signer/context-module/src/shared/model/ClearSignContext.ts [hunk 2]
```diff
@@ -12,31 +12,66 @@ export enum ClearSignContextType {
...
-    };
+  : Type extends ClearSignContextType.TRUSTED_NAME
+    ? {
+        type: ClearSignContextType.TRUSTED_NAME;
+        valuePath: GenericPath;
+        types: string[];
+        sources: string[];
+      }
+    : {
+        type: ClearSignContextType.TOKEN | ClearSignContextType.NFT;
```
Comment (jiyuzhuang) [inline packages/signer/context-module/src/shared/model/ClearSignContext.ts:37]: [MAYBE] Use parameter `Type` to get a concrete `ClearSignContextType`
```suggestion
        type: Type;
```
This will give out 
```ts
{ 
  type: ClearSignContextType.TOKEN;
  valuePath:  GenericPath;
}
```
if you pass `ClearSignContextType.TOKEN` as generic parameter which is more clear, I think. 
But this is not perfect including those two previous `extends` keywords, for example, the type 
```ts
ClearSignContextReference<ClearSignContextType.ENUM | ClearSignContextType.TRUSTED_NAME>
``` 
will give out 
```ts
{
  type: ClearSignContextType.ENUM;
  valuePath: GenericPath;
  id: number;
}
```
where you wanna get is the union of the two. Perhaps this does exist in use cases, but the typing here is not perfect narrowing the types.

Comment (aussedatlo) [reply packages/signer/context-module/src/shared/model/ClearSignContext.ts:37]: You are right, will add it 👍

**Comment by aussedatlo** [comment]
> LGTM generally, but some small updates are needed to let the CI pass as well as the place of `[DSDK-601]` in the description.

Should be fixed now, thx !

**Review by jiyuzhuang — APPROVED** [review]

File: packages/signer/context-module/src/transaction/data/HttpTransactionDataSource.ts [hunk 3]
```diff
@@ -43,7 +46,7 @@ export class HttpTransactionDataSource implements TransactionDataSource {
     address,
     selector,
   }: GetTransactionDescriptorsParams): Promise<
-    Either<Error, ClearSignContextSuccess[]>
+    Either<Error, ClearSignContext[]>
```
Comment (paoun-ledger) [inline packages/signer/context-module/src/transaction/data/HttpTransactionDataSource.ts:49]: out of curiosity, why did you change the return type ? If I'm not mistaken, you never return an ERROR in a Right return?

File: packages/signer/context-module/src/transaction/data/HttpTransactionDataSource.ts [hunk 4]
```diff
@@ -227,8 +247,32 @@ export class HttpTransactionDataSource implements TransactionDataSource {
...
+      typeof calldata === "object" &&
+      Object.entries(calldata).every(
+        ([id, values]) =>
+          typeof id === "string" &&
+          typeof values === "object" &&
+          Object.entries<CalldataTransactionDescriptor>(values).every(
+            ([value, obj]) =>
+              typeof value === "string" &&
+              typeof obj === "object" &&
+              obj.signatures !== undefined &&
```
Comment (paoun-ledger) [inline packages/signer/context-module/src/transaction/data/HttpTransactionDataSource.ts:264]: maybe you should verify that `typeof obj.data === "string"`?

File: packages/signer/context-module/src/transaction/data/HttpTransactionDataSource.ts (hunk 3 above)
Comment (aussedatlo) [reply packages/signer/context-module/src/transaction/data/HttpTransactionDataSource.ts:49]: It's not needed, It's a rest of one of my test so I'll remove it 👍

**Review by paoun-ledger — APPROVED** [review]
